### PR TITLE
[Storage] Add `list_blob_names` API to Blob SDK

### PR DESCRIPTION
**Pull request by jalauzon-msft**
In response to #19755

Adding a new API to `ContainerClient` called `list_blob_names` which will call the List Blobs API but only deserialize and return the names of the blobs. This should be much faster than the existing `list_blobs`.

Initial perf test results comparing regular `list_blobs` to `list_blob_names`:
|            | Throughput - Ops/s |           |             |
|------------|--------------------|-----------|-------------|
| Blob Count | list_blobs | list_blob_names | Diff  |
| 5          | 255.7              | 367.1     | 1.435666797 |
| 500        | 8.556              | 75.88     | 8.868630201 |
| 50000      | 0.0708             | 0.9809    | 13.85451977 |

### CODE REVIEW

File: sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py [hunk 1]
```diff
@@ -785,6 +796,39 @@ def list_blobs(self, name_starts_with=None, include=None, **kwargs):
             command, prefix=name_starts_with, results_per_page=results_per_page,
             page_iterator_class=BlobPropertiesPaged)
 
+    @distributed_trace
+    def list_blob_names(self, **kwargs: Any) -> ItemPaged[str]:
+        """Returns a generator to list the names of blobs under the specified container.
+        The generator will lazily follow the continuation tokens returned by
+        the service.
+
+        :keyword str name_starts_with:
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py:805]: Do we need to support the `include` keyword param?

**Review by annatisch — APPROVED** [review]
This is really awesome - thanks!

**Comment by annatisch** [comment]
Adding @mikeharder as an FYI - @jalauzon-msft has updated the Storage list_blobs perf test to optionally switch to the "list names only" API which has big perf improvements.
Not sure there a way we can add that to the pipeline...?

**Comment by mikeharder** [comment]
> Adding @mikeharder as an FYI - @jalauzon-msft has updated the Storage list_blobs perf test to optionally switch to the "list names only" API which has big perf improvements. Not sure there a way we can add that to the pipeline...?

We can certainly add these tests to the pipeline eventually, so we can compare listing names only with full metadata.  There are few complications:

1. If we run this perf test in the pipeline, it will fail against the GA package since it doesn't contain this API.  I think we can live with this, the perf framework should gracefully handle failures.  We'll get results for sources but a failure for the GA package.

2. Currently, we run the same set of tests across all languages, as defined in https://github.com/Azure/azure-sdk-tools/blob/main/tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/tests.yml.  There are hacky ways we could add a test for just python, but the long-term fix is for me to split tests.yml into one file per package per language: 
 https://github.com/Azure/azure-sdk-tools/issues/3530.  I will move this issue to the top of my backlog, and start by enabling pipelines to specify their own tests.yml, so we can slowly migrate pipelines to the new model.

File: sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py (hunk 1 above)
Comment (jalauzon-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py:805]: As discussed offline, will leave this out for now.

**Pull request merged**